### ISSUE TITLE
Product OrderBy NameAsc & NameDesc not working properly

### ISSUE THREAD

**Issue by brakim** [nopSolutions/nopCommerce#6651]
nopCommerce version: 4.60

Steps to reproduce the problem: 

Mix products with and without translation.

Products with translation will be sortet first by A-Z, then products without.

![image](https://user-images.githubusercontent.com/894222/225245373-acb4fa20-258b-4aaf-86eb-00edb46553bd.png)

```
var query =
                    from product in productsQuery
                    join localizedProperty in localizedPropertyRepository.Table on new
                        {
                            product.Id,
                            languageId = currentLanguageId,
                            keyGroup = nameof(Product),
                            key = nameof(Product.Name)
                        }
                        equals new
                        {
                            Id = localizedProperty.EntityId,
                            languageId = localizedProperty.LanguageId,
                            keyGroup = localizedProperty.LocaleKeyGroup,
                            key = localizedProperty.LocaleKey
                        } into localizedProperties
                    from localizedProperty in localizedProperties.DefaultIfEmpty(new LocalizedProperty { LocaleValue = product.Name })
                    select new { localizedProperty, product };

                if (orderBy == ProductSortingEnum.NameAsc)
                    productsQuery = from item in query
                        orderby item.localizedProperty.LocaleValue, item.product.Name
                        select item.product;
                else
                    productsQuery = from item in query
                        orderby item.localizedProperty.LocaleValue descending, item.product.Name descending
                        select item.product;

                return productsQuery;
```

translates to 

```
ORDER BY
	[localizedProperties].[LocaleValue],
	[p].[Name],
	[pc_2].[DisplayOrder]
```

My fix

```
var query =
                    from product in productsQuery
                    join localizedProperty in localizedPropertyRepository.Table on new
                        {
                            product.Id,
                            languageId = currentLanguageId,
                            keyGroup = nameof(Product),
                            key = nameof(Product.Name)
                        }
                        equals new
                        {
                            Id = localizedProperty.EntityId,
                            languageId = localizedProperty.LanguageId,
                            keyGroup = localizedProperty.LocaleKeyGroup,
                            key = localizedProperty.LocaleKey
                        } into localizedProperties
                    from localizedProperty in localizedProperties.DefaultIfEmpty()
                    select new
                    {
                        sortName = localizedProperty == null ? product.Name : localizedProperty.LocaleValue, 
                        product
                    };

                if (orderBy == ProductSortingEnum.NameAsc)
                    productsQuery = from item in query
                        orderby item.sortName
                        select item.product;
                else
                    productsQuery = from item in query
                        orderby item.sortName descending
                        select item.product;

                return productsQuery;
```

Translates better
```
ORDER BY
	IIF([localizedProperties].[Id] IS NULL, [p].[Name], [localizedProperties].[LocaleValue]),
	[pc_2].[DisplayOrder],
	[p].[Name]
```

The problem lies here, because that does not translate right to sql.

**Comment by kuldeepNageshwar17** [comment]
Hi  @AndreiMaz  ,
 I am interested in this discussion and investigation of how this is my first time starting open-sourcing on this project can you please help me to create and let me know how can I move further on this?

**Comment by exileDev** [comment]
Closed #6651
